src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -585,9 +585,9 @@ private void handleSubscribeResponse(SubscribeResponse res) throws SubscriptionN
         }
     }
 
-    private void handleUnsubcribeResponse(UnsubscribeResponse res) throws SubscriptionNotFoundException {
+    private void handleUnsubscribeResponse(UnsubscribeResponse res) throws SubscriptionNotFoundException {
         Subscription sub = subs.get(res.getStreamId(), res.getStreamPartition());
-        sub.setState(Subscription.State.UNSUBSCRIBED);
```

Comment:
We should probably keep this line! (The application using the library might hold a reference to the `Subscription` instance, as it is returned from the `subscribe` method. So even though we clean the instance up from `subs`, we should update it to reflect the correct state.)

Reply:
Thank you, I'll revert that line